debezium-core/src/main/java/io/debezium/connector/common/Partition.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.common;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Describes the source partition to be processed by the connector in connector-specific terms
+ * and provides its representation as a Kafka Connect source partition.
+ */
+public interface Partition {
+    Map<String, String> getSourcePartition();
+
+    /**
+     * Implementations provide a collection of connector-specific partitions based on the connector task configuration.
+     */
+    interface Provider<P extends Partition> {
+        Collection<P> getPartitions();
```

Comment:
Should this rather return `List` or `Set`?  I think it's a good idea to be lenient for parameter types but more authoritative in terms of provided guarantees for return value types.

Reply:
I tried to keep it as simple/permissive as in `org.apache.kafka.connect.storage.OffsetStorageReader#offsets(Collection<Map<String, T>> partitions)`. But I agree the return type could be more restrictive. `Set` seems better suitable since we want the result to contain only unique values.